### PR TITLE
SLL9: Java enablement

### PR DESCRIPTION
**Pull request by avshiliaev**
## What does this PR change?

SUSE Liberty Linux 9 support, java enablement. 

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- Unit tests were added

- [x] **DONE**

## Links

Implements https://github.com/SUSE/spacewalk/issues/19158

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql" (Test skipped, there are no changes to test)
- [ ] Re-run test "java_pgsql_tests"         
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [x] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/suse/manager/reactor/test/dummy_grains_sll.json [hunk 1]
```diff
@@ -0,0 +1,194 @@
+{
+  "kernel": "Linux",
+  "domain": "",
+  "zmqversion": "4.0.8",
+  "kernelrelease": "4.1.27-27-default",
+  "pythonpath": [
+    "/usr/bin",
+    "/usr/lib64/python27.zip",
+    "/usr/lib64/python2.7",
+    "/usr/lib64/python2.7/plat-linux2",
```
Comment (cbosdo) [inline java/code/src/com/suse/manager/reactor/test/dummy_grains_sll.json:10]: Would be better to get grains from a real SLL9

File: java/code/src/com/suse/manager/reactor/utils/test/dummy_packages_redhatprodinfo_centos_sll.json [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    "name": "cat /etc/centos-release",
+    "start_time": "10:30:07.679765",
+    "result": true,
+    "duration": 22.903,
+    "__run_num__": 1,
+    "changes": {
+      "pid": 15301,
+      "retcode": 0,
+      "stderr": "",
+      "stdout": "CentOS Stream release 9.0 (Name)"
```
Comment (mcalmer) [inline java/code/src/com/suse/manager/reactor/utils/test/dummy_packages_redhatprodinfo_centos_sll.json:35]: It's only a test, but we do not support CentOS 9 (Stream). SLL needs to come together with something else than CentOS. Maybe Rocky or Alma. I am not 100% sure what is a supported variant. Maybe "standalone" SLL is now also an option.

File: java/spacewalk-java.changes [hunk 3]
```diff
@@ -1,3 +1,4 @@
+- Add SUSE Liberty Linux support for RHEL9 and CentOS Stream 9 based clients
```
Comment (mcalmer) [inline java/spacewalk-java.changes:1]: "CentOS Stream 9" is not supported by SUSE Manager. We should not name it in a changelog

**Review by mcalmer — CHANGES_REQUESTED** [review]

**Review by cbosdo — COMMENTED** [review]
Looks good to me, but you could probably factorize some code to avoid duplicate code

**Review by mcalmer — APPROVED** [review]

File: java/code/src/com/suse/manager/reactor/utils/test/dummy_packages_redhatprodinfo_centos_sll.json (hunk 2 above)
Comment (admd) [reply java/code/src/com/suse/manager/reactor/utils/test/dummy_packages_redhatprodinfo_centos_sll.json]: @mcalmer is right.

I just checked with Liberty team and they mentioned that as a part of Liberty they support RHEL/CentOS(except CentOS stream), and Oracle Linux upon request.

**Comment by avshiliaev** [comment]
@cbosdo I need some help here. I get [this linter error](https://github.com/uyuni-project/uyuni/actions/runs/3513142635/jobs/5885625400). It's not the code I've touched here and I'm not quite sure what needs to be changed :smile:

File: java/code/src/com/suse/manager/reactor/utils/test/RhelUtilsTest.java [hunk 4]
```diff
@@ -342,34 +379,66 @@ public void testDetectRhelProductRocky() throws Exception {
...
+        return createExpandedSupportChannel(user, version, archLabel, channelLabel,
+                "SLL", "SUSE Liberty Linux");
+    }
+
+    public static Channel createExpandedSupportChannel(User user, String version,
+                                                       String shortName, String friendlyName) {
+        try {
+            return createExpandedSupportChannel(user, version, "x86_64", UUID.randomUUID().toString(),
+                    shortName, friendlyName);
+        } catch (Exception e) {
```
Comment (cbosdo) [inline java/code/src/com/suse/manager/reactor/utils/test/RhelUtilsTest.java:411]: you need a newline after `}`

**Review by cbosdo — APPROVED** [review]
LGTM. There is a failure in the unit test, likely not related to your PR as I saw the same on one of mine. It would be better if someone fixes it before merging. I also would like your PR to be merged before https://github.com/uyuni-project/uyuni/pull/6186

**Comment by avshiliaev** [comment]
Alright, thank you :rocket:

**Pull request merged**